### PR TITLE
introduce GET /consent/v2

### PR DESCRIPTION
**Pull request by arthuradr**

### CODE REVIEW

File: consent.yaml [hunk 1]
```diff
@@ -185,6 +185,260 @@ paths:
...
+                            description: JSON schema of customer data
+                            purpose: explain why custom data is needed
+                            properties:
+                              employerName:
+                                type: string
+                                displayText: Employer name
+                                uiComponent:
+                                  inputType: text
+                                rules:
+                                  required: Employer name is required
```
Comment (patrwongadr) [inline consent.yaml]: this example is showing up as an error for me in stoplight:
image

our model says it should be a `string` type but the example shows it's an object:
image

File: consent.yaml [hunk 2]
```diff
@@ -1444,3 +1713,13 @@ components:
...
+    QueryParamPageSize:
+      name: pageSize
+      in: query
+      required: false
+      schema:
+        type: integer
+        minimum: 1
+        default: 1000
+        example: 1000
+      description: Page size to request. Default is 1000
```
Comment (patrwongadr) [inline consent.yaml]: this looks like a dupe of https://github.com/Adatree/api-specs/pull/47/files#diff-fa21ef9cc73e0385ed1e3d52420c3d0871b0b4880678a5e40eebec4d18b11543R1660-R1668

File: consent.yaml [hunk 3]
```diff
@@ -185,6 +185,260 @@ paths:
...
+          example: 148c9187-e52b-431f-9b24-7225e75ca2be
+          required: false
+          schema:
+            type: string
+        - name: cdrArrangementId
+          in: query
+          example: 45db4977-b12f-42bf-808d-2680eb4bd366
+          required: false
+          schema:
+            type: string
```
Comment (patrwongadr) [inline consent.yaml]: this already has a `Parameter` defined https://github.com/Adatree/api-specs/pull/47/files#diff-fa21ef9cc73e0385ed1e3d52420c3d0871b0b4880678a5e40eebec4d18b11543R1677-R1684

File: consent.yaml (hunk 1 above)
Comment (arthuradr) [reply consent.yaml]: Having a look now, it is so hard to please everyone 🤣  
https://validator.swagger.io was happy with https://raw.githubusercontent.com/Adatree/api-specs/97788fc6ebe47b6baa598453a672e7345461cc9d/consent.yaml

File: consent.yaml [hunk 4]
```diff
@@ -185,6 +185,260 @@ paths:
...
+          example: hello@adatree.com.au
+          required: false
+          schema:
+            type: string
+        - name: consentId
+          in: query
+          example: 148c9187-e52b-431f-9b24-7225e75ca2be
+          required: false
+          schema:
+            type: string
```
Comment (patrwongadr) [inline consent.yaml]: this one too https://github.com/Adatree/api-specs/pull/47/files#diff-fa21ef9cc73e0385ed1e3d52420c3d0871b0b4880678a5e40eebec4d18b11543R1685-R1692

File: consent.yaml (hunk 1 above)
Comment (patrwongadr) [reply consent.yaml]: i'm guessing `validator.swagger.io` didn't validate the example against the model 🤣

File: consent.yaml (hunk 2 above)
Comment (arthuradr) [reply consent.yaml]: different default parameter... :(

File: consent.yaml (hunk 1 above)
Comment (arthuradr) [reply consent.yaml]: is it fixed now?

**Review by patrwongadr — APPROVED** [review]

**Pull request merged**